### ISSUE TITLE
Could not establish trust relationship for the SSL/TLS secure channel

### ISSUE THREAD

**Issue by sebwerner** [Microsoft/IIS.WebManager#505]
I am getting the following error when opening the IIS extension in Server Manager:

Unknown error has occurred when trying to initialize Web Server Module: "The underlying connection was closed: Could not establish trust relationship for the SSL/TLS secure channel."

**Comment by atomsmith** [comment]
For me, this error was caused by expiration of the self-signed certificates installed by the [IIS.Administration](https://github.com/microsoft/IIS.Administration) component. Uninstalling and reinstalling this component created fresh certificates that are good for another two years. However, it looks like [IIS.Administration is end-of-life](https://github.com/microsoft/IIS.Administration/issues/305). Does that mean this project is dead by extension?

**Comment by DumboJetEngine** [comment]
I have the same problem.
I have uninstalled and reinstalled "IIS Management Scripts and Tools" from the server roles, restarted, but nothing helped.
My localhost certificate is expired, and thus the certificate of https://server:55539/ .
I have also tried removing the certificate and that has completely broken the IIS Management API.
Any ideas are welcome.
How can I set the certificate manually to a specific certificate in the WebHosting store?

**Comment by yaqiyang** [comment]
The "IIS Management Scripts and Tools" server role does not help. That is not the IIS Management tool used in WAC. 

Can you access https://server:55539/ directly? If it works, the certificate is fine. If not, 
*  Reinstall IIS Administration application on the server from https://github.com/microsoft/IIS.Administration/releases/tag/v6.0.0. That should update the certificate,
*  If re-installing IIS Admin does not help, you might not have enough permissions to access it from the WAC.  You need to give yourself permission by doing https://github.com/microsoft/IIS.Administration#:~:text=If%20you%20still%20don%27t%20have%20permissions

**Comment by DumboJetEngine** [comment]
Originally, I could access  https://server:55539/ , but I was getting a certificate error and I had to bypass this on the browser (which is expected). The expiration date on the certificate was on a past date.
But the IIS extension in WAC did not work at all, and I was getting the mentioned error.
Later, I removed a "localhost" certificate from the server the above API stopped working all together.

Thanks for your help.
I have managed to fix the problem now.
I have uninstalled "IIS Administration" from the "Installed Apps" addon/section in WAC (Windows Admin Center).
Uninstalling using the linked installer and an `/uninstall` flag did not work for some reason and it was leaving things behind that prevented a successful reinstall.
After that, installing fixed everything! :)
